### ISSUE TITLE
Variables names on datasummary_crosstab

### ISSUE THREAD

**Issue by caleb-kwon** [vincentarelbundock/modelsummary#455]
Hi Vincent,

Thanks for the marvelous package! A quick inquiry regarding datasummary_crosstab and variable names. I know that datasummary allows one to make custom names for variables, but I'm not sure this feature works on datasummary_crosstab. 

Code below:

```

library(modelsummary)
library(tables)
url <- 'https://vincentarelbundock.github.io/Rdatasets/csv/palmerpenguins/penguins.csv'
penguins <- read.csv(url)

# Name change within function: Gives Error 
datasummary_crosstab((`Hello What` = species) ~ sex * island, data = penguins)

# Name change prior: Gives Error 
penguins$`Hello What` = penguins$species
datasummary_crosstab(`Hello What` ~ sex * island, data = penguins)

# Using datasummary: Works 
datasummary(`Hello What` ~ N, data = penguins)

```

If this was intentional, is there a workaround? 

Thanks!

**Comment by vincentarelbundock** [comment]
Thanks a lot for the report @caleb-kwon ! 

This looks like a bug to me, but I'm not 100% sure what the source is. My guess is it has to do with the way we "combine" formulas internally, which might make us lose the backticks.

@eldersb may have a better idea since he played with the code most recently. If he doesn't, I'll look into this as soon as I find some time.

**Comment by vincentarelbundock** [comment]
#### TLDR

I just pushed a new version to Github which allows 1 of your 3 attempts to work.

#### Details

Thanks again for the report. I looked into this a bit and it is going to be hard to support renaming on the fly in the `datasummary_crosstab()` formula. The problem is that this table is so customized that we essentially deconstruct the formula completely and rebuild it based on the raw variable names in `all.vars()`. To support renaming on the fly, we would need much smarter parsing of the formula. A “robust” strategy to do that is possible, but it will require a considerable effort.

#### Solution

In the meantime, I made changes to allow using backticks in the formula, which allows users to rename their variables in the data.frame before calling `datasummary_crosstab()`.

Using the most recent Github version, you can do:

``` r
library(modelsummary)

dat <- mtcars
dat$`# of Cylinders` <- dat$cyl
datasummary_crosstab(`# of Cylinders` ~ am * vs, data = dat)
```

![cyl](https://user-images.githubusercontent.com/987057/161391002-0966ed94-4489-4f09-ae97-d12396553cbe.PNG)

#### Informative error

The other options now raise an informative error:

``` r
datasummary_crosstab((`# of Cylinders` = cyl) ~ am * vs, data = dat)
#> Error: The `formula` argument of the `datasummary_crosstab` function does not support the `=` sign or the `Heading()` function. You can rename variables in the data frame before calling `datasummary_crosstab` and use backticks to enclose variable names with spaces. For example:
#> 
#> dat <- mtcars
#> dat$`# of Cylinders` <- dat$cyl
#> datasummary_crosstab(`# of Cylinders` ~ am * gear, data = dat)
#> 
#> Note that the `datasummary()` function supports the `=` sign and the `Heading()` function to rename variables. If you would like to contribute code to support those in `datasummary_crosstab`, please visit the `modelsummary` development website: https://github.com/vincentarelbundock/modelsummary
```

**Comment by caleb-kwon** [comment]
Wow. Amazing! Thanks for the explanation and fix. This is great!

Thanks again for all your help and the package. The package has been an incredible addition to my workflow.
